### DISCUSSION TITLE
Error during install.

### DISCUSSION THREAD

**Discussion by xxedgexx** [Keyfactor/ejbca-ce#262] [General]
I can't seem to figure out what's causing this:

     [java] Exception in thread "main" java.util.ServiceConfigurationError: org.ejbca.ui.cli.infrastructure.command.CliCommandPlugin: Provider org.ejbca.ui.cli.keybind.InternalKeyBindingCreateCommand could not be instantiated
     [java] 	at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:582)
     [java] 	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:804)
     [java] 	at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:722)
     [java] 	at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1395)
     [java] 	at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.<init>(CommandLibrary.java:58)
     [java] 	at org.ejbca.ui.cli.infrastructure.library.CommandLibrary.<clinit>(CommandLibrary.java:39)
     [java] 	at org.ejbca.ui.cli.EjbcaEjbCli.main(EjbcaEjbCli.java:32)
     [java] Caused by: jakarta.ejb.NoSuchEJBException: EJBCLIENT000079: Unable to discover destination for request for EJB StatelessEJBLocator for "ejbca/cesecore-ejb/InternalKeyBindingMgmtSessionBean", view is interface org.cesecore.keybind.InternalKeyBindingMgmtSessionRemote, affinity is None
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:620)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.protocol.remote.RemotingEJBClientInterceptor.handleInvocationResult(RemotingEJBClientInterceptor.java:57)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.TransactionPostDiscoveryInterceptor.handleInvocationResult(TransactionPostDiscoveryInterceptor.java:148)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.DiscoveryEJBClientInterceptor.handleInvocationResult(DiscoveryEJBClientInterceptor.java:130)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.NamingEJBClientInterceptor.handleInvocationResult(NamingEJBClientInterceptor.java:87)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.call(AuthenticationContextEJBClientInterceptor.java:59)
     [java] 	at org.jboss.ejb.client.AuthenticationContextEJBClientInterceptor.handleInvocationResult(AuthenticationContextEJBClientInterceptor.java:52)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.TransactionInterceptor.handleInvocationResult(TransactionInterceptor.java:212)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:622)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.getResult(EJBClientInvocationContext.java:551)
     [java] 	at org.jboss.ejb.client.EJBClientInvocationContext.awaitResponse(EJBClientInvocationContext.java:1003)
     [java] 	at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:182)
     [java] 	at org.jboss.ejb.client.EJBInvocationHandler.invoke(EJBInvocationHandler.java:116)
     [java] 	at com.sun.proxy.$Proxy6.getAvailableTypesAndProperties(Unknown Source)
     [java] 	at org.ejbca.ui.cli.keybind.InternalKeyBindingCreateCommand.<init>(InternalKeyBindingCreateCommand.java:71)
     [java] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
     [java] 	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
     [java] 	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
     [java] 	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
     [java] 	at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:780)
     [java] 	... 5 more
     [java] 	Suppressed: org.jboss.remoting3.ServiceOpenException: Unknown service name jboss.ejb
     [java] 		at org.jboss.remoting3.remote.RemoteReadListener.handleEvent(RemoteReadListener.java:445)
     [java] 		at org.jboss.remoting3.remote.RemoteReadListener.handleEvent(RemoteReadListener.java:51)
     [java] 		at org.xnio.ChannelListeners.invokeChannelListener(ChannelListeners.java:92)
     [java] 		at org.xnio.conduits.ReadReadyHandler$ChannelListenerHandler.readReady(ReadReadyHandler.java:66)
     [java] 		at org.xnio.nio.NioSocketConduit.handleReady(NioSocketConduit.java:89)
     [java] 		at org.xnio.nio.WorkerThread.run(WorkerThread.java:591)

BUILD FAILED
/opt/ejbca-ce/build.xml:72: The following error occurred while executing this line:
/opt/ejbca-ce/bin/cli.xml:98: The following error occurred while executing this line:
/opt/ejbca-ce/bin/cli.xml:126: The following error occurred while executing this line:
/opt/ejbca-ce/bin/cli.xml:527: Java returned: 1

Total time: 8 seconds

I checked the db credentials.  It's all working as far as the db is concerned.  This is wildfly 27...  Any ideas?  Thanks.

**Comment by primetomas** [comment]
WildFly 27 is Jakarta EE 10 _only_. Meaning that it is a compatibility breaking release of WildFly. EJBCA is still Jakarta EE 8 and 9. EE 10 support is in the works, but it will be a hard cut-over, once EE 10 support is there, you will not be able to run on older WildFly/JBoss any longer. We haven't seen such backwards-compatibility breaking updates before with Java Enterprise, so this is an unusual situation.

So for now, you need to stick to WildFly 26.

Your error means that EJBCA did not deploy correctly, so WildFly with EJBCA could not start up, hence it is unreachable which the error says. More information about the actual error you will find in the WildFly server.log.
https://doc.primekey.com/ejbca/troubleshooting-guide
